tensorboard/program.py

```diff
@@ -413,12 +422,15 @@ def __init__(self, wsgi_app, flags):
     host = flags.host
     port = flags.port
 
-    # Without an explicit host, we default to serving on all interfaces,
-    # and will attempt to serve both IPv4 and IPv6 traffic through one
-    # socket.
-    self._auto_wildcard = not host
+    self._auto_wildcard = flags.bind_all
     if self._auto_wildcard:
+      # Serve on all interfaces, and attempt to serve both IPv4 and IPv6
+      # traffic through one socket.
       host = self._get_wildcard_address(port)
+    elif host is None:
+      host = 'localhost'
+
+    self._host = host
```

Comment:
Is `self._host` actually used anywhere?  Later down you read `self.host`, which is the property set by werkzeug:
https://github.com/pallets/werkzeug/blob/bd60d52ba14f32a38caffc674fb17c9090ef70ce/src/werkzeug/serving.py#L728

Not sure if those are intentionally different or a typo.

Reply:
Good catch; this was an oversight. I’ll conservatively use `self._host`.